src/nav_channel.cpp

```diff
@@ -0,0 +1,221 @@
+#include <ros/ros.h>
+#include <nav_channel/TaskStatus.h>
+#include <nav_channel/TaskGoalPosition.h>
+#include <nav_channel/Task.h>
+#include <string>
+//#include <prop_mapper/PropArray.h> // isn't included yet
+#include <nav_channel/PropArray.h> // fake prop map to replace prop_mapper for now
+#include <geometry_msgs/PoseWithCovarianceStamped.h>
+#include <nav_msgs/Odometry.h>
+
+class NavChannel {
+public:
+    NavChannel(): nh("")
+    {
+        // task_to_exec will confirm we are on nav_channel, then execute task
+        task_to_exec = nh.subscribe("task_to_execute", 10, &NavChannel::navChannelCallback, this);
+        
+        // props will get the prop map and store it
+        prop_map = nh.subscribe("props", 10, &NavChannel::propMapCallback, this);
+
+        // global_position/local grabs position from mavros and stores it
+        global_pos = nh.subscribe("mavros/global_position/local", 10, &NavChannel::globalPositionCallback, this);
```

Comment:
Maybe we should use local_position/pose instead of global because we are going to set the destination with setpoint_position/local... thoughts? setpoint_position/local uses [this](http://docs.ros.org/en/api/geometry_msgs/html/msg/PoseStamped.html) message type.

Reply:
NVM, the IQ sim package uses global position, I don't really know if it matters  which one we use I am 95% they are roughly the same. Just something to keep in mind if we seem to be having issues.